src/js/components/stock-movement-wizard/request/AddItemsPage.jsx

```diff
@@ -1287,16 +1285,31 @@ class AddItemsPage extends Component {
   saveAndExit(formValues) {
     const errors = this.validate(formValues).lineItems;
     if (!errors.length) {
-      this.saveRequisitionItemsInCurrentStepWithAlert({
-        lineItems: formValues.lineItems,
-        callback: () => {
-          let redirectTo = '/openboxes/stockMovement/list?direction=INBOUND';
-          if (!this.props.supportedActivities.includes('MANAGE_INVENTORY') && this.props.supportedActivities.includes('SUBMIT_REQUEST')) {
-            redirectTo = '/openboxes/dashboard';
-          }
-          window.location = redirectTo;
-        },
-      });
+      const lineItems = _.filter(formValues.lineItems, item => !_.isEmpty(item));
+      const zeroedLines = _.some(lineItems, item => !item.quantityRequested || item.quantityRequested === '0');
+      if (zeroedLines || this.state.isRequestFromWard) {
+        this.confirmSave(() => this.saveRequisitionItemsInCurrentStepWithAlert({
```

Comment:
I think you could use the `saveRequisitionItemsInCurrentStep` or at least rename `saveRequisitionItemsInCurrentStepWithAlert` if you want to use a callback and you don't want to refactor the `saveRequisitionItemsInCurrentStep`. The `saveRequisitionItemsInCurrentStepWithAlert` naming is no longer valid because there is no `alert` anymore inside.

Reply:
@kchelstowski actually you don't need it at all just make it as:
```
this.props.showSpinner();
    return this.saveRequisitionItemsInCurrentStep(lineItems)
      .then(res =>  {
        let redirectTo = '/openboxes/stockMovement/list?direction=INBOUND';
        if (!this.props.supportedActivities.includes('MANAGE_INVENTORY') && this.props.supportedActivities.includes('SUBMIT_REQUEST')) {
              redirectTo = '/openboxes/dashboard';
        }
        window.location = redirectTo;
      })
      .catch(() => {
        this.props.hideSpinner();
      });
```